streaming/src/main/scala/org/apache/spark/streaming/dstream/InputDStream.scala

```diff
@@ -47,6 +49,27 @@ abstract class InputDStream[T: ClassTag] (@transient ssc_ : StreamingContext)
   /** This is an unique identifier for the input stream. */
   val id = ssc.getNewInputStreamId()
 
+  /**
+   * A rate estimator configured by the user to compute a dynamic ingestion bound for this stream.
+   * @see `RateEstimator`
+   */
+  protected [streaming] val rateEstimator = newEstimator()
+
+  /**
+   * Return the configured estimator, or `noop` if none was specified.
+   */
+  private def newEstimator() =
```

Comment:
Again, this can be all within the RateController, does not need to be a top level field in DStream. In fact it can be something on a `RateEstimator` objects.